### DISCUSSION TITLE
Unable to connect to LAN devices

### DISCUSSION THREAD

**Discussion by LLL151** [rustdesk/rustdesk#10363] [General]
### Bug Description

When I connect the LAN devices, it will say  Failed to connect to rs-ny.rustdesk.com:21116.
If i use LAN IP the connect. it will say Failed to connect to 192.168.1.110:21118.
This happens whether it is Windows, Linux, or Android.
I am currently using 1.3.3, and the same situation occurs when I change both sides to 1.3.6.

### How to Reproduce

Use Windows 11 23H2  connect other LAN devices
Use Windows MSTSC No problem

Ubuntu 20.04 Terminal output：
(base) robot@robot-BATTLE-AX-B760M-E-WIFI-V20:~$ lsof -i :21116
(base) robot@robot-BATTLE-AX-B760M-E-WIFI-V20:~$ 
(base) robot@robot-BATTLE-AX-B760M-E-WIFI-V20:~$ lsof -i :21118
(base) robot@robot-BATTLE-AX-B760M-E-WIFI-V20:~$ 

### Expected Behavior

It will connect all the LAN devices

### Operating system(s) on local (controlling) side and remote (controlled) side

Windows 11 23H2 -> ubuntu 20.04

### RustDesk Version(s) on local (controlling) side and remote (controlled) side

1.3.6 -> 1.3.6

### Screenshots

![image](https://github.com/user-attachments/assets/144ba338-bf3a-4126-a3b3-3d7589258d4b)
![image](https://github.com/user-attachments/assets/4906f8ec-b487-4a46-9aa7-b591989ec907)

**Comment by xlionjuan** [comment]
You need to enable Direct IP connect if you want to use it

**Reply by LLL151** [reply]
Well, I don't know why this option is now available. A week ago, I didn't set anything and could just click it and it would work without enabling Direct IP connect. Now I have to enter the LAN IP to connect. And the device I clicked on was already discovered. Since the device has been discovered on the LAN, the LAN should be used first. And this login seemed to identify me as a new connected device, causing me to re-enter the password.
![image](https://github.com/user-attachments/assets/341ba325-7db6-436d-afec-d1dd2a9cfbc1)

**Reply by xlionjuan** [reply]
Because public servers are always **overload**.
@LLL151

**Reply by LLL151** [reply]
It has nothing to do with the public server. What I mean is that since I discovered the device in the LAN, I should prioritize identifying and using the LAN IP for device connection instead of using the public server.

**Reply by xlionjuan** [reply]
> It has nothing to do with the public server. 

**WRONG**, it is not how RustDesk works, RustDesk still ***need*** to communicate to rendezvous server when establishing connections, **even if both side can be direct connected**.

And, the online indicator is also rely on rendezvous server.

**Reply by LLL151** [reply]
This can explain why the online status is still wrong even if I have connected using the LAN IP. 
However, if all connections have to go through the rendezvous server, what is the point of LAN discovery? I can install it on other devices and of course get the ID and password of the corresponding device.

**Reply by xlionjuan** [reply]
LAN IP's online status can not show because it can be the security problem (leak)

The reason why ID connect is more preferred is because both clients **trusts** rendezvous server, no matter it is ACLs, rate limit or keys.

**Comment by rustdesk** [comment]
Close and lock since meaningless discussions

**Comment by rustdesk** [comment]
@xlionjuan Thanks for your patience and kindness, though @LLL151 is not the one you can convince, :(
